core/eth/utils.go

```diff
@@ -126,3 +128,9 @@ func bitmapToBytesArray(bitmap *big.Int) []byte {
 	}
 	return bytesArray
 }
+
+func isZeroValuedReservation(reservation paymentvault.IPaymentVaultReservation) bool {
+	zeroReservation := paymentvault.IPaymentVaultReservation{}
+
+	return reflect.DeepEqual(reservation, zeroReservation)
```

Comment:
`maps.Equal` for performance

Reply:
I had trouble with this because `IPaymentVaultReservation` is not a map. is it more performant to put the struct into a map and use `maps.Equal`?

Reply:
Ah it's a struct. We could compare each field individually (or i'm fine with using reflect as well).

Do you know if eth rpc will return nil vs. empty byte slice for `[]byte` type fields? 
If they return empty byte slice, this comparison will evaluate to false